department-of-veterans-affairs/va.gov-cms
# VACMS-14565: Make content release operations on nodes without original versions safer.

A somewhat different approach to @swirtSJW's in #14567, though incorporating some of his changes.

This makes the change in a different place and adds some tests that I think will provide more confidence that these errors won't cause problems during normal editing operations 😩 .

## QA steps (Copied from #14567)

As an admin
1. Go to a [Abilene Vet center facility](https://pr14569-rbbisnso6sf3muduwsd8ozkb4orvvv85.ci.cms.va.gov/abilene-vet-center) and delete it.
   - [ ] Validate that it deletes.
2. Go to the migration and run it as an update migration (check the "update" check box)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/0e0b349a-4b3a-4c26-8699-31e0f38b9c2d)  

   - [ ] validate that the vet center you deleted, was recreated by [going here](https://pr14569-rbbisnso6sf3muduwsd8ozkb4orvvv85.ci.cms.va.gov/abilene-vet-center) 
   - [ ] Open a new tab for [dblog](https://pr14569-rbbisnso6sf3muduwsd8ozkb4orvvv85.ci.cms.va.gov/admin/reports/dblog)  validate there are no related errors/warnings in watchdog
3. Edit a facility node and give it status and publish it.
   - [ ] validate in dblog that a content release should have been triggered.
4. Edit the same node but do not change its status. and publish and save the revision
   - [ ] validate in dblog that a content release should **not** have been triggered
5. Edit the same node and change the status, save and publish it.
   - [ ] Validate in dblog that a content release should have been triggered.

6. Edit the same node, make a change to status but save as draft
    - [ ] validate that no release trigger took place.
7. Edit edit the same node and change to publish
   - [ ] validate that a release trigger took place.
